cli/src/commands/auth.rs

```diff
@@ -29,30 +41,46 @@ async fn handle_auth_login(
 ) -> Result<()> {
     let api_uri = &config.connection.uri;
     let ignore_certs = config.ignore_certs();
-    config.auth_info =
-        PhylumApi::login(config.auth_info, ignore_certs, api_uri, matches.get_flag("reauth"))
-            .await?;
+    config.auth_info = PhylumApi::login(
+        config.auth_info,
+        matches.get_one("token-name").cloned(),
+        ignore_certs,
+        api_uri,
+        matches.get_flag("reauth"),
+    )
+    .await?;
     save_config(config_path, &config).map_err(|error| anyhow!(error))?;
     Ok(())
 }
 
 /// Display the current authentication status to the user.
 pub async fn handle_auth_status(config: Config, timeout: Option<u64>) -> CommandResult {
-    if config.auth_info.offline_access().is_none() {
-        print_user_warning!("User is not currently authenticated");
-        return Ok(ExitCode::NotAuthenticated);
-    }
+    let auth_type = match config.auth_info.offline_access() {
+        Some(token) if is_locksmith_token(token) => "API key",
+        Some(_) => "OpenID Connect",
+        None => {
+            print_user_warning!("User is not currently authenticated");
+            return Ok(ExitCode::NotAuthenticated);
+        },
+    };
 
     // Create a client with our auth token attached.
     let api = PhylumApi::new(config, timeout).await?;
 
     let user_info = api.user_info().await;
 
+    debug!("User info reponse: {:?}", user_info);
+
     match user_info {
         Ok(user) => {
             print_user_success!(
-                "Currently authenticated as '{}' with long lived refresh token",
-                user.email
+                "Currently authenticated as '{}<{}>' via {}",
+                user.name.map_or(" ".into(), |mut n| {
+                    n.push(' ');
+                    n
+                }),
```

Comment:
If I interpret this correctly, this should output `...as ' <email>' via...` if `user.name` is `None`.
Wouldn't that be equivalent to having `...as '{} <{}>' via...` in the format string + `unwrap_or` on the field? Maybe an empty string was intended in the first argument?

Also it's super minor and I'm not sure it is not already getting optimized away anyway, but considering that `"".into()` allocates a string, I'd use `map_or_else` instead.

```suggestion
                user.name.map_or_else(|| "".into(), |mut n| {
                    n.push(' ');
                    n
                }),
```

Reply:
Yep. This should have been empty string... I've updated the code to use `Default::default()` instead.